src/lib.rs

```diff
@@ -1050,6 +1050,31 @@ impl<T> Drop for Receiver<T> {
                 // SAFETY: see safety comment at top of function
                 unsafe { dealloc(self.channel_ptr) };
             }
+            // The sender has observed the RECEIVING state and is currently reading the waker from
+            // a poll. We need to loop here until we observe the MESSAGE or DISCONNECTED state.
+            // We busy loop here since we know the sender is done very soon.
+            #[cfg(any(feature = "std", feature = "async"))]
+            UNPARKING => loop {
+                hint::spin_loop();
+                // ORDERING: The load above has already synchronized with the write of the message.
+                match channel.state.load(Relaxed) {
+                    MESSAGE => {
+                        // SAFETY: we are in the message state so the message is initialized
+                        unsafe { channel.drop_message() };
+
+                        // SAFETY: see safety comment at top of function
+                        unsafe { dealloc(self.channel_ptr) };
+                        break;
+                    }
+                    DISCONNECTED => {
+                        // SAFETY: see safety comment at top of function
+                        unsafe { dealloc(self.channel_ptr) };
```

Comment:
The loop always results in a `dealloc` eventually. Instead of deallocating in all branches just before the `break`, can we deallocate after the loop instead? I'm guessing compiling with optimizations will join these anyway, but this could help reduce compiled code size if it is not, and I don't think it would hurt readability.